daemon/cmd/cni.go

```diff
@@ -174,19 +228,18 @@ func installCNIConfFile(opts *option.DaemonConfig) error {
 		}
 	}
 
-	// commit CNI config
-	if err := renameio.WriteFile(path.Join(confDir, filename), contents, 0644); err != nil {
-		return fmt.Errorf("failed to write CNI configuration file at %s: %w", opts.WriteCNIConfigurationWhenReady, err)
-	}
-	log.Infof("Wrote CNI configuration file to %s", opts.WriteCNIConfigurationWhenReady)
+	dest := path.Join(confDir, filename)
 
-	// Remove the old non-conflist cilium file
-	if filename != "05-cilium.conf" {
-		_ = os.Remove(path.Join(confDir, "05-cilium.conf"))
-	}
-	// Rename any non-cilium CNI config files.
-	if opts.CNIExclusive {
-		cleanupOtherCNI(confDir, filename)
+	// Check to see if existing file is the same; if so, do nothing
+	existingContents, err := os.ReadFile(dest)
+	if err == nil && bytes.Equal(existingContents, contents) {
+		log.Debugf("Existing CNI configuration file %s unchanged", dest)
+	} else {
```

Comment:
this might obscure errors that aren't just missing file, might make sense to log cases where the error is not `ErrNotExist`?

Reply:
Sure; though the error is uninteresting (it's just a shortcut), so I figure there's no need to pile up more ignorable errors in the log.

Reply:
Added a log line.